services/dynamic-help-service.ts

```diff
@@ -0,0 +1,26 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+import os = require("os");
+
+export class DynamicHelpService implements IDynamicHelpService {
+	constructor(private $injector: IInjector) { }
+	public isProjectType(...args: string[]): IFuture<boolean> {
+		return ((): boolean => {
+			// resolve here as Project.IProject is not part of common-lib
+			var $project: Project.IProject = this.$injector.resolve("project");
```

Comment:
My idea is to use dynamicHelpService in tns as well, because its help also has some os specific text. So instead of having DynamicHelpService on two places, I've decided to place it here and note that one of the methods is not usable for tns. In fact if we change this method this way:
```Typescript
public isProjectType(...args: string[]): IFuture<boolean> {
  var $project = this.$injector.resolve("project");
  if($project && $project.getProjectDir && $project.getProjectDir().wait()) {
	var framework = $project.projectData.Framework.toLowerCase();
        return _.any(args, arg => arg.toLowerCase() === framework);
  }

   return true;
}
```
it could be called inside tns without throwing error. It will just return true.